### ISSUE TITLE
Low scores of yolov8

### ISSUE THREAD

**Issue by EdisionWew** [ultralytics/ultralytics#1687]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi，thanks for jobs. I got models by yolov8s and yolov5m with a same dataset. But I found that objects of yolov8 got a lower score than yolov5m. Could you help me to explain it?

**Comment by glenn-jocher** [comment]
The difference in detection scores between YOLOv8s and YOLOv5m could be attributed to several factors, such as differences in network architecture, training data, or hyperparameters used during training. However, it is difficult to provide a definitive answer without more information on your specific use case and dataset.

One possible explanation is that YOLOv8s uses a slightly lower object detection threshold by default than YOLOv5m, which could result in fewer false positives but also fewer true positives. Nonetheless, both models should perform similarly on new datasets and images, so it is recommended to evaluate their performance across multiple datasets and images to get a better understanding of their strengths and weaknesses.

If you need more detailed help on this issue, I recommend providing more context and details such as your training parameters, dataset, and specific performance metrics. You can also review the documentation at https://docs.ultralytics.com for more information on YOLOv8 and how it differs from YOLOv5.

**Comment by sarpx** [comment]
Make sure you are using the latest version of version 8. I had the same situation and I saw agument features that should not be in weights and they removed them in the last version. If you are using the first versions of version 8, I strongly recommend you to update.

https://github.com/ultralytics/ultralytics/issues/1194

**Comment by glenn-jocher** [comment]
That's a great tip! In general, it's always recommended to use the latest version of any software to ensure that you are benefiting from the latest updates and bug fixes. So, for anyone facing similar issues with YOLOv8, updating to the latest version should definitely be the first step. Also, make sure to carefully read the release notes and documentation to understand any changes or new features that have been introduced. If you continue to have issues after updating, feel free to open a new issue on the YOLOv8 GitHub repository or consult the official documentation at https://docs.ultralytics.com.

**Comment by EdisionWew** [comment]
> 

Hi, I observed that the weight of cls loss in default yaml of yolov5m was more higher than box loss, which is different from yolov8s. And I did the experiment that I made the weight of cls loss increased by a factor of ten and box loss remain unchanged, It showes that the object scores of yolov8s get higer than before and the mAP of model remains unchanged. 
![image](https://user-images.githubusercontent.com/23101204/229399479-7250267c-7526-48c2-86f6-6fc393c6edb4.png)
The left image is the weight of cls loss increase by a factor of ten, and the right image is the result of weight of cls loss by default. 

I wonder why the parameters are set this way.

**Comment by sarpx** [comment]
@EdisyonWew
Can you share the args.yaml files from both tutorials?
These files are located in the runs folder in the training outputs.

**Comment by EdisionWew** [comment]
> @EdisyonWew Can you share the args.yaml files from both tutorials? These files are located in the runs folder in the training outputs.

Sure. 
yolov8s with cls=5.0
`task: detect
mode: train
model: runs/detect/train38/weights/best.pt
data: ./yolodataset_18bs_26ks_yolov8s.yaml
epochs: 100
patience: 50
batch: 128
imgsz: 640
save: true
save_period: -1
cache: false
device:
- 0
- 1
- 2
- 3
workers: 6
project: null
name: null
exist_ok: false
pretrained: true
optimizer: SGD
verbose: true
seed: 0
deterministic: true
single_cls: false
image_weights: false
rect: false
cos_lr: false
close_mosaic: 10
resume: false
overlap_mask: true
mask_ratio: 4
dropout: 0.0
val: false
split: val
save_json: false
save_hybrid: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
show: false
save_txt: false
save_conf: false
save_crop: false
hide_labels: false
hide_conf: false
vid_stride: 1
line_thickness: 3
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
boxes: true
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: false
opset: null
workspace: 4
nms: false
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 5.0
dfl: 1.5
fl_gamma: 0.0
label_smoothing: 0.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
cfg: null
v5loader: false
tracker: botsort.yaml
save_dir: runs/detect/train40`

 **and yolov8s by default.yaml**
`task: detect
mode: train
model: /data01/erwei.wang/yolov8s.pt
data: ./yolodataset_18bs_26ks_yolov8s.yaml
epochs: 300
patience: 50
batch: 128
imgsz: 640
save: true
save_period: -1
cache: false
device:
- 0
- 1
- 2
- 3
workers: 6
project: null
name: null
exist_ok: false
pretrained: true
optimizer: SGD
verbose: true
seed: 0
deterministic: true
single_cls: false
image_weights: false
rect: false
cos_lr: false
close_mosaic: 10
resume: false
overlap_mask: true
mask_ratio: 4
dropout: 0.0
val: false
split: val
save_json: false
save_hybrid: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
show: false
save_txt: false
save_conf: false
save_crop: false
hide_labels: false
hide_conf: false
vid_stride: 1
line_thickness: 3
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
boxes: true
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: false
opset: null
workspace: 4
nms: false
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
fl_gamma: 0.0
label_smoothing: 0.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
cfg: null
v5loader: false
tracker: botsort.yaml
save_dir: runs/detect/train38`

**Comment by sarpx** [comment]
Your settings are pretty much the same but the only difference seems to be the number of epochs. You had training for 100 periods in one and 300 periods in the other. Sometimes, the results may decrease as it tends to find more similarities when it is excessively memorized. Please make the same epochs of the training you think you got bad results and try it with the other one.

**Comment by EdisionWew** [comment]
> Your settings are pretty much the same but the only difference seems to be the number of epochs. You had training for 100 periods in one and 300 periods in the other. Sometimes, the results may decrease as it tends to find more similarities when it is excessively memorized. Please make the same epochs of the training you think you got bad results and try it with the other one.

Yeah，I set epoch much smaller than before， but the loss curves of iterator both  converged. So it has little difference of models. A big difference of yaml above is the weight cls loss of "cls:5" and "cls: 0.5". It showes that make  the weight of cls loss increase to get object scores higher.

**Comment by glenn-jocher** [comment]
@EdisionWew sure, I can help explain with concepts and ideas. In YOLOv8s and YOLOv5m, there are several hyperparameters that can be adjusted during training to tweak the model's performance. One of these is the weighting of the classification loss (cls) and bounding box regression loss (box). By increasing the weight of cls loss, the model becomes more confident in its object detection scores, which can lead to fewer false positives at the expense of potentially fewer true positives. 

However, it's important to note that changing hyperparameters like these might not always lead to better results. It's essential to evaluate the model's performance on multiple datasets and images to get a more comprehensive understanding of its strengths and weaknesses. Additionally, sometimes increasing the training epochs can help reduce overfitting and increase accuracy, as the model has more time to learn the dataset-specific features.
